rpcs3/Emu/Audio/Cubeb/CubebBackend.cpp

```diff
@@ -358,18 +391,45 @@ std::tuple<cubeb_devid, std::string, u32> CubebBackend::GetDefaultDeviceAlt(Audi
 
 	if (out_dev_name.empty())
 	{
+		Cubeb.notice("No default device available");
 		return {};
 	}
 
 	return GetDevice(out_dev_name);
 }
 
-long CubebBackend::data_cb(cubeb_stream* /* stream */, void* user_ptr, void const* /* input_buffer */, void* output_buffer, long nframes)
+long CubebBackend::data_cb(cubeb_stream* stream, void* user_ptr, void const* /* input_buffer */, void* output_buffer, long nframes)
 {
+	if (nframes <= 0)
+	{
+		Cubeb.error("data_cb called with nframes=%d", nframes);
+		return 0;
+	}
+
+	if (!output_buffer)
+	{
+		Cubeb.error("data_cb called with invalid output_buffer");
+		return CUBEB_ERROR;
```

Comment:
If you return error here, then you must also handle state update in state_cb callback

Reply:
how so?
I'd assume the state will simply be error ?
Remember this is just a sanity check. I doubt that this will happen unless something is really wrong.

Reply:
According to docs stream will be drained. So the state would probably be CUBEB_STATE_DRAINED.

Reply:
I can't find the proper docs anymore.
Maybe you can propose some code since I completely forgot what this is even about 😂